db/indexes.go

```diff
@@ -195,16 +267,19 @@ func waitForIndexes(bucket *base.CouchbaseBucketGoCB, useXattrs bool) error {
 	return nil
 }
 
-// Issues adhoc consistency=request_plus query to determine if specified is ready.  Retries on timeout.
+// Issues adhoc consistency=request_plus query to determine if specified is ready.  Retries indefinitely on timeout, backoff retry on indexer error.
 func waitForIndex(bucket *base.CouchbaseBucketGoCB, indexName string, queryStatement string) error {
 
 	for {
 		_, err := bucket.Query(queryStatement, nil, gocb.RequestPlus, true)
 		// Retry on timeout error, otherwise return
-		if err == nil || err != base.ErrViewTimeoutError {
-			return err
-		} else {
+		if err == nil {
+			return nil
+		}
+		if err == base.ErrViewTimeoutError {
```

Comment:
The next line might need log redaction for the bucket name (or is this pending in another PR?)